apps/api/src/app/shared/framework/idempotency.e2e.ts

```diff
@@ -222,6 +225,41 @@ describe('Idempotency Test', async () => {
         })
       );
     });
+
+    describe('Allowed Authentication Security Schemes', () => {
+      it('should set Idempotency-Key header when ApiKey security scheme is used to authenticate', async () => {
+        const key = '10';
+        const { headers } = await session.testAgent
+          .post(path)
+          .set(HttpResponseHeaderKeysEnum.IDEMPOTENCY_KEY, key)
+          .set('authorization', `ApiKey ${session.apiKey}`)
+          .send({ data: 201 })
+          .expect(201);
+        expect(headers[HttpResponseHeaderKeysEnum.IDEMPOTENCY_KEY.toLowerCase()]).to.exist;
+      });
+
+      it('should set rate limit headers when a Bearer security scheme is used to authenticate', async () => {
+        const key = '10';
+        const { headers } = await session.testAgent
+          .post(path)
+          .set(HttpResponseHeaderKeysEnum.IDEMPOTENCY_KEY, key)
+          .set('authorization', session.token)
+          .send({ data: 201 })
+          .expect(201);
+        expect(headers[HttpResponseHeaderKeysEnum.IDEMPOTENCY_KEY.toLowerCase()]).to.exist;
+      });
+
+      it('should NOT set rate limit headers when NO authorization header is present', async () => {
+        const key = '10';
+        const { headers } = await session.testAgent
+          .post(path)
+          .set(HttpResponseHeaderKeysEnum.IDEMPOTENCY_KEY, key)
+          .set('authorization', '')
+          .send({ data: 201 })
+          .expect(401);
+        expect(headers[HttpResponseHeaderKeysEnum.IDEMPOTENCY_KEY.toLowerCase()]).not.to.exist;
+      });
+    });
```

Comment:
Adding tests to verify that API Idempotency is only enabled for desired authentication schemes.